src/libexpr/value-to-xml.cc

```diff
@@ -118,14 +118,19 @@ static void printValueAsXML(EvalState & state, bool strict, bool location,
 
             break;
 
-        case tList1: case tList2: case tListN: {
+        case nList: {
             XMLOpenElement _(doc, "list");
             for (unsigned int n = 0; n < v.listSize(); ++n)
                 printValueAsXML(state, strict, location, *v.listElems()[n], doc, context, drvsSeen);
             break;
         }
 
-        case tLambda: {
+        case nFunction: {
+            if (!v.isLambda()) {
+                // TODO: Serialize primops and other functions
+                doc.writeEmptyElement("unevaluated");
```

Comment:
This `NormalType` change actually discovered a small bug in the xml conversion: it prints `unevaluated` for `tPrimOp` and `tPrimOpApp`'s. I didn't want to bother changing the format of that so I left it.